### ISSUE TITLE
"type-fest": "4.7.0", `PartialDeep` break down 4.6.0 works fine

### ISSUE THREAD

**Issue by tianyingchun** [sindresorhus/type-fest#742]
```ts
export type ConfigEnvBase = {};
export type UserConfigFn<UserConfig, ConfigEnv extends ConfigEnvBase> = (env: ConfigEnv) => UserConfig | Promise<UserConfig>;
export type UserConfigExport<UserConfig, ConfigEnv extends ConfigEnvBase = ConfigEnvBase> = UserConfig | Promise<UserConfig> | UserConfigFn<UserConfig, ConfigEnv>;
export declare const defineConfig: <UserConfig, ConfigEnv extends ConfigEnvBase = ConfigEnvBase>(config: UserConfigExport<UserConfig, ConfigEnv>) => UserConfigExport<UserConfig, ConfigEnv>;

import { type PartialDeep } from 'type-fest';
export interface FlatForgeOptions {
    /**
     * the project root directory, this thould be optional
     */
    projectCwd: string;
}
type DefineConfigFn = (userConfig: PartialDeep<FlatForgeOptions>) => UserConfigExport<PartialDeep<FlatForgeOptions>>;
export declare const defineConfig: DefineConfigFn;

defineConfig({
    projectCwd: 'src',
})
```

<img width="675" alt="image" src="https://github.com/sindresorhus/type-fest/assets/2833251/9c338633-d4e5-4c2a-9ca0-a4e4ad736caa">

## Upvote & Fund

- We're using [Polar.sh](https://polar.sh/sindresorhus) so you can upvote and help fund this issue.
- The funding will be given to active contributors.
- Thank you in advance for helping prioritize & fund our backlog.

<a href="https://polar.sh/sindresorhus/type-fest/issues/742">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/sindresorhus/type-fest/issues/742/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/sindresorhus/type-fest/issues/742/pledge.svg">
</picture>
</a>

below is 4.6.0

<img width="646" alt="image" src="https://github.com/sindresorhus/type-fest/assets/2833251/b6c9fc83-1b8c-4529-90fa-b2b4d51b90ea">

**Comment by Emiyaaaaa** [comment]
https://github.com/sindresorhus/type-fest/pull/738/files

`interface` is not `UnknownRecord` but `type` is, I'm sorry about mistake this.

@sindresorhus

**Comment by sindresorhus** [comment]
@Emiyaaaaa Reverted for now. It should not recurse into prototype properties though, so I assume we just need a better fix for that?


Also note: https://github.com/sindresorhus/type-fest/pull/738#discussion_r1387299139

**Comment by Emiyaaaaa** [comment]
reopened issue #651, let's talk there

**Comment by Emiyaaaaa** [comment]
It‘s been fixed at 4.7.1
